### PR TITLE
Development

### PR DESCRIPTION
**Pull request by llama2474**

### CODE REVIEW

File: CMP1903_A1_2324/Game.cs [hunk 1]
```diff
@@ -4,11 +4,100 @@
 using System.Text;
 using System.Threading.Tasks;
 
+namespace CMP1903_A1_2324
```
Comment (calmac04) [inline CMP1903_A1_2324/Game.cs]: you have seemingly placed copies of all the other classes into the game class making it not run

**Review by calmac04 — COMMENTED** [review]
your development code does work however i had to make numerous changes to have it functioning. first you have other classes code within game which causes it no to run. second in your testing class (which is not in the pull request) you are missing the line
using Systems.Diagnostics;
to allow for the use of Debug.Assert() and this must be added among the using lines to have the code functional.
other than those two issues the code runs correctly and fulfills all the requirements laid out by the assignment. additionally you may want to attempt the optional extra for game and allow for continuous rolls

**Comment by d-hmlton** [comment]
New commits fixed the issues with code not running 👍

File: CMP1903_A1_2324/Game.cs [hunk 2]
```diff
@@ -8,7 +8,63 @@ namespace CMP1903_A1_2324
...
+            return ConsoleResponse;
+        }
+        private void Response(string Response) //Forces the reply to be Y or N to continue
+        {
+         bool Valid = false;
+           while (Valid == false) //While loop forces a correct response to continue
+            {
+             if (Response.ToUpper() != "Y" & Response.ToUpper() != "N") //Checks if the response is Y or N
+              {
+                    Console.WriteLine("Response not valid");
```
Comment (d-hmlton) [inline CMP1903_A1_2324/Game.cs:57]: If input is invalid here (not Y/N), the code gets trapped in an infinite loop of printing "Response not valid"

**Review by d-hmlton — COMMENTED** [review]
Code runs without issues and meets the criteria defined, using Debug.Assert() well to test the code! The exception handling for the Y/N input should be modified to prompt for an additional input or end code - currently results in an infinite loop - beyond that didn't find other bugs.
For further improvement, you could add XML comments to your classes as described in 1st criteria on the CRG. _("XML documenting comments are throughout the code.")_ Well done! 👍

**Pull request merged**